### PR TITLE
Fix type completion for attribute tokens

### PR DESCRIPTION
**Pull request by MartinGC94**
# PR Summary
Fixes tab completion for the type name in attribute tokens like: `[Parameter<Tab>()]`

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs [hunk 1]
```diff
@@ -416,7 +416,7 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                     case TokenKind.Generic:
                     case TokenKind.MinusMinus: // for native commands '--'
                     case TokenKind.Identifier:
-                        if (tokenAtCursor.TokenFlags != TokenFlags.TypeName)
+                        if ((tokenAtCursor.TokenFlags & TokenFlags.TypeName) == 0)
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs:419]:
```suggestion
                        if (!tokenAtCursor.TokenFlags.HasFlag(TokenFlags.TypeName))
```

File: src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs [hunk 2]
```diff
@@ -960,7 +960,7 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                     }
                 }
                 
-                if (typeNameToComplete is null && tokenAtCursor?.TokenFlags == TokenFlags.TypeName)
+                if (typeNameToComplete is null && tokenAtCursor is not null && (tokenAtCursor.TokenFlags & TokenFlags.TypeName) != 0)
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs:963]:
```suggestion
                if (typeNameToComplete is null && tokenAtCursor?.TokenFlags.HasFlag(TokenFlags.TypeName) == true)
```

**Review by daxian-dbw — COMMENTED** [review]
Looks good. Left 2 comments.

File: src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs (hunk 1 above)
Comment (MartinGC94) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs:419]: Nice, I didn't realize they had made HasFlag fast.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**